dizoo/dmc2gym/envs/dmc2gym_env.py

```diff
@@ -0,0 +1,207 @@
+
+
+def dmc2gym_action_space(dim, minimum=-1, maximum=1, dtype=np.float32) -> Box:
+    return Box(np.repeat(minimum, dim).astype(dtype), np.repeat(maximum, dim).astype(dtype), dtype=dtype)
+
+
+def dmc2gym_reward_space(minimum=0, maximum=1, dtype=np.float32) -> Callable:
+
+    def reward_space(frame_skip=1) -> Box:
+        return Box(
+            np.repeat(minimum * frame_skip, 1).astype(dtype),
+            np.repeat(maximum * frame_skip, 1).astype(dtype),
+            dtype=dtype
+        )
+
+    return reward_space
+
+
+dmc2gym_env_info = {
+    "ball_in_cup": {
+        "catch": {
+            "observation_space": dmc2gym_observation_space(8),
+            "state_space": dmc2gym_state_space(8),
+            "action_space": dmc2gym_action_space(2),
+            "reward_space": dmc2gym_reward_space()
+        }
+    },
+    "cartpole": {
+        "balance": {
+            "observation_space": dmc2gym_observation_space(8),
+            "state_space": dmc2gym_state_space(8),
+            "action_space": dmc2gym_action_space(1),
+            "reward_space": dmc2gym_reward_space()
+        },
+        "swingup": {
+            "observation_space": dmc2gym_observation_space(8),
+            "state_space": dmc2gym_state_space(8),
+            "action_space": dmc2gym_action_space(1),
+            "reward_space": dmc2gym_reward_space()
+        }
+    },
+    "cheetah": {
+        "run": {
+            "observation_space": dmc2gym_observation_space(17),
+            "state_space": dmc2gym_state_space(17),
+            "action_space": dmc2gym_action_space(6),
+            "reward_space": dmc2gym_reward_space()
+        }
+    },
+    "finger": {
+        "spin": {
+            "observation_space": dmc2gym_observation_space(9),
+            "state_space": dmc2gym_state_space(9),
+            "action_space": dmc2gym_action_space(1),
+            "reward_space": dmc2gym_reward_space()
+        }
+    },
+    "reacher": {
+        "easy": {
+            "observation_space": dmc2gym_observation_space(6),
+            "state_space": dmc2gym_state_space(6),
+            "action_space": dmc2gym_action_space(2),
+            "reward_space": dmc2gym_reward_space()
+        }
+    },
+    "walker": {
+        "walk": {
+            "observation_space": dmc2gym_observation_space(24),
+            "state_space": dmc2gym_state_space(24),
+            "action_space": dmc2gym_action_space(6),
+            "reward_space": dmc2gym_reward_space()
+        }
+    }
+}
+
+
+@ENV_REGISTRY.register('dmc2gym')
+class DMC2GymEnv(BaseEnv):
+
+    def __init__(self, cfg: dict = {}) -> None:
+        assert cfg.domain_name in dmc2gym_env_info, '{}/{}'.format(cfg.domain_name, dmc2gym_env_info.keys())
+        assert cfg.task_name in dmc2gym_env_info[
+            cfg.domain_name], '{}/{}'.format(cfg.task_name, dmc2gym_env_info[cfg.domain_name].keys())
+
+        self._domain_name = cfg.domain_name
+        self._task_name = cfg.task_name
+        if "frame_skip" in cfg:
```

Comment:
use default_config and deep_merge_dicts to deal with this case

Reply:
Thanks, it's a very good suggestion!